cmd/jag/commands/flash.go

```diff
@@ -5,18 +5,108 @@
 package commands
 
 import (
+	"bufio"
 	"bytes"
 	"fmt"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strconv"
+	"strings"
 
 	"github.com/google/uuid"
 	"github.com/spf13/cobra"
 	"github.com/toitlang/jaguar/cmd/jag/directory"
 )
 
+// Returns position, size of the partitions in partitions.csv
+func getPartitions(toitToolchainPath string, positions map[string]int, sizes map[string]int) {
+	// Load default partitions positions, that can be overridden by the
+	// partitions.csv file.  Some of these (bootloader, partitions) are
+	// commented out in the partitions.csv because they can't be changed.
+	positions["bootloader"] = 0x1000
+	positions["partitions"] = 0x8000
+	positions["ota"] = 0xd000
+	positions["ota_0"] = 0x10000
+	sizes["bootloader"] = 0x7000
+	sizes["partitions"] = 0x0c00
+	COLUMN_NAME := 0
+	// COLUMN_TYPE := 1
+	COLUMN_SUBTYPE := 2
+	COLUMN_POSITION := 3
+	COLUMN_SIZE := 4
+
+	file, err := os.Open(filepath.Join(toitToolchainPath, "partitions.csv"))
+	if err != nil {
+		fmt.Println("**Warning: Could not find partitions.csv, using default partition sizes.")
+		return
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		partitionType := ""
+		line := scanner.Text()
+		comment := strings.Index(line, "#")
+		if comment != -1 {
+			line = line[:comment]
+		}
+		line = strings.TrimSpace(line)
+		if line != "" {
+			fields := strings.Split(line, ",")
+			maxIndex := -1
+			for index, field := range fields {
+				maxIndex = index
+				field = strings.TrimSpace(field)
+				if index == COLUMN_NAME {
+					partitionType = field
+				} else if index == COLUMN_SUBTYPE {
+					if field != "" {
+						partitionType = field
+					}
+				} else if index == COLUMN_POSITION || index == COLUMN_SIZE {
+					num, err := strconv.ParseInt(field, 0, 32)
+					if err != nil || partitionType == "" {
+						fmt.Println("**Warning: Could not parse number in partitions.csv")
+					} else {
+						if index == COLUMN_POSITION {
+							positions[partitionType] = int(num)
+						} else {
+							sizes[partitionType] = int(num)
+						}
+					}
+				}
+			}
+			if maxIndex < COLUMN_SIZE {
+				fmt.Println("**Warning: Could not parse line in partitions.csv (missing fields)")
+			}
+		}
+	}
+}
+
+func hex(num int) string {
+	return fmt.Sprintf("0x%x", num)
+}
+
+func createZapBytesFile(sizes map[string]int, name string) (*os.File, error) {
+	// Create a file with zap bytes (0xff) for clearing the OTA data partition.
```

Comment:
Update comment now that this is also used for NVS.

```suggestion
	// Create a file with zap bytes (0xff) for clearing select partitions.
```